### DISCUSSION TITLE
Does Avalonia automatically support universal lanuages (automatic native fallback fonts for all languages) like native solutions (eg. Maui)? Or is this inherently not possible for Avalonia?

### DISCUSSION THREAD

**Discussion by jonmdev** [AvaloniaUI/Avalonia#14835] [Q&A] [answered]
In .NET MAUI, we have automatic support of all languages the OS supports in things like TextEditors and Labels, because they are native elements that can work everywhere on earth (Microsoft/Google/Apple make sure of it).

I suspect this is not possible given Avalonia renders its own UI. When I used Unity's UI Toolkit similarly, it could not do this for the same reason as I suspect will be the case for Avalonia.

I suspect that any system that renders its own UI will be dependent on embedded fonts, and thus if you wanted every language on earth you would have to embed all these fonts and have localization rules for fallback orders of font usage, etc. (like I presume the native OS does).

Is this true? 

ie. Does a stock version of Avalonia automatically support Hindi, Chinese, Korean, English, etc if input into a text field or label?

I am currently working in MAUI and I would love to switch to Avalonia, but easy worldwide language support is important, so I may not have a choice.

Thanks for any clarification.

**Comment by Gillibald** [comment]
We support all writing systems

**Reply by jonmdev** [reply]
I am not sure what that means? Specifically:

- Do you automatically utilize the native multi-language fonts built into the OS?
- Or do you need to embed each language font into the app and set up font fallback rules to traverse them?

Thanks again.

**Reply by Gillibald** [reply]
We have proper font fallback. Fonts are automatically selected depending on the needed unicode characters.

**Reply by jonmdev** [reply]
Okay, but do you need to embed every language as a font that you want coverage for? Or does Avalonia automatically access the native built in OS fonts to do this? Is there any documentation on how this works?

Thanks again.

**Reply by timunie** [reply]
depends on your OS. In Browser for example you are in a sandbox and we can't access system fonts afaik. On Windows it should "just work".

**Reply by jonmdev** [reply]
Do you know about Android or iOS? Do fonts automatically fall back to OS fonts? Eg. If you copy and past some Chinese or Korean into an app you have made, will it display even if you have not embedded an appropriate font to display it?

Thanks.

**Reply by Gillibald** [reply, marked as answer]
Yes only WASM has a limited font manager because the environment is sandboxed

**Comment by jonmdev** [comment]
Very nice to see this document explaining the IME support on Avalonia.

https://dev.to/avalonia/turning-it-up-to-11-34jn

I will consider this a "YES" overall then, as I need multilanguage support on Android/iOS without embedding 50+ MB of fonts.

Thanks.
